tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -1378,4 +1384,29 @@ class Restrictions {
 
     return classDefinitions;
   }
+
+  ClassDefinition? _getParentClass(ClassDefinition currentClass) {
+    if (currentClass.extendsClass is! ResolvedInheritanceDefinition) {
+      return null;
+    }
+
+    return (currentClass.extendsClass as ResolvedInheritanceDefinition)
+        .classDefinition;
+  }
+
+  ClassDefinition? _findOtherTableClassInHierarchy(
```

Comment:
Suggestion: rename to `_findTableClassInParentClasses` or `_findTableClassInSuperClasses`.

Just to clarify we are only checking our parents and never our children or siblings.